firmware/examples/TaskScheduler/source/main.cpp

```diff
@@ -0,0 +1,61 @@
+#include <cstdarg>
+
+#include "L0_LowLevel/uart0.hpp"
+#include "L2_Utilities/debug_print.hpp"
+#include "L4_Application/task.hpp"
+
+class PrinterTask : public rtos::Task<1024>
+{
+ public:
+    bool Setup() override
+    {
+        DEBUG_PRINT("Setup() for: %s", kName);
+        return true;
+    };
+    bool PreRun() override
+    {
+        DEBUG_PRINT("PreRun() for: %s", kName);
+        return true;
+    };
+    bool Run() override
+    {
+        run_count_ += 1;
+        DEBUG_PRINT("%s: %ld", message_, run_count_);
+        if (run_count_ == 10)
+        {
+            Delete();
+        }
+        return true;
+    };
+    PrinterTask(const char * task_name, const char * message)
+        : Task(task_name, rtos::Priority::kMedium),
+          message_(message),
+          run_count_(0){};
+
+ private:
+    const char * message_;
+    uint32_t run_count_;
+};
+
+int main()
+{
+    DEBUG_PRINT("Starting TaskScheduler example...");
+    PrinterTask * printer_one =
```

Comment:
Could you have an example of you doing this without using new? 
You could put that version of the variable in global space so it has dedicated memory for it in the .data section.

Reply:
Without the new, a hard fault occurs.

Reply:
Ok. Will look past this for this commit. Can you add an issue in the issues section, as well as a TODO in your code stating that we should move away from the usage of new here when we find a fix.